test/e2e/network/loadbalancer.go

```diff
@@ -188,9 +188,6 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 		err = tcpJig.CheckServiceReachability(ctx, tcpService, execPod)
 		framework.ExpectNoError(err)
 
-		ginkgo.By("hitting the TCP service's NodePort")
-		e2eservice.TestReachableHTTP(ctx, nodeIP, tcpNodePort, e2eservice.KubeProxyLagTimeout)
-
```

Comment:
> it seems that the CI somehow managed to provide connectivity between the worker running the tests and the Nodes on the cluster under test.
>
> However, this is not always true, causing these test to fail.
>
> For NodePort tests, we can test the functionality within the cluster, no need to test NodePorts from the e2e.test binary.


OK. So the reason is that majority of tests are running tests using public clusters, where nodes have public IPs.
Our large scale scalability tests are using private clusters, where nodes have only private IPs, which obviously means that these won't be reachable.

Back then we implemented the functionality of "bastion" (there is ~1 node that has public IP via which we can ssh to nodes).
I'm not sure if we can somehow reuse this concept here, but just pointing you to it, to double check if we can't reuse it.

Reply:
Forgot to add a link:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/ssh/ssh.go#L53

Reply:
We try to keep the e2e test as much hermetic as possible, in this case NodePorts can be tested within the cluster so I prefer to avoid to test them also from an external place, that has the dependencies on a lot of external factors like Firewalls, LoadBalancer, ....

For LoadBalancers the test is realistic, since you expect your application to be reachable externally

Reply:
>We try to keep the e2e test as much hermetic as possible,

+1

Can you point me to tests that already exercise this functionality? [Just to double check we are not missing anything subtle here...]

Reply:
yeah, NodePort are Conformance and they have also another tests like these that exercise them under the hood

https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20master&include-filter-by-regex=NodePort